osu.Game/Screens/OnlinePlay/Lounge/Components/RoomPanel.cs

```diff
@@ -204,10 +208,27 @@ private void load(OverlayColourProvider colours)
                                                                 Direction = FillDirection.Vertical,
                                                                 Children = new Drawable[]
                                                                 {
-                                                                    roomName = new TruncatingSpriteText
+                                                                    roomNameFlow = new FillFlowContainer
                                                                     {
                                                                         RelativeSizeAxes = Axes.X,
-                                                                        Font = OsuFont.GetFont(size: 28)
+                                                                        AutoSizeAxes = Axes.Y,
+                                                                        Direction = FillDirection.Horizontal,
+                                                                        Children = new Drawable[]
+                                                                        {
+                                                                            roomName = new TruncatingSpriteText
+                                                                            {
+                                                                                Anchor = Anchor.BottomLeft,
+                                                                                Origin = Anchor.BottomLeft,
+                                                                                Font = OsuFont.GetFont(size: 28),
+                                                                            },
+                                                                            linkButton = new ExternalLinkButton(formatRoomUrl(Room.RoomID ?? 0))
```

Comment:
Falling back to 0 if the room ID is not there is an interesting choice, I'd choose not showing the button instead.

Reply:
It's a temporary number, I've made the button not show at all when a room ID is not present (see alpha assignment below this line).